plugins/TTAnalyzer.cc

```diff
@@ -5,104 +7,449 @@
+        diBJet_DPhi[wp + "-Pt"] = 0;
+        diBJet_DEta[wp + "-Pt"] = 0;
+      }
+      selectedJets_diBJetExcluded[wp + "-Pt"] = std::vector<uint8_t>();
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID_DRcut.size(); ijet++){
+        if(m_diBJet_PtChosen[wp].idxs.first != ijet && m_diBJet_PtChosen[wp].idxs.second != ijet)
+          selectedJets_diBJetExcluded[wp + "-Pt"].push_back(ijet);
+      }
+
+      // CSVv2-ordered
+      diBJet_p4[wp + "-CSVv2"] = m_diBJet_CSVv2Chosen[wp].p4;
+      diBJet_idxs[wp + "-CSVv2"] = std::make_pair(m_diBJet_CSVv2Chosen[wp].idxs.first, m_diBJet_CSVv2Chosen[wp].idxs.second);
+      if(m_diBJet_CSVv2Chosen[wp].idxs.first >=0 ){
+        diBJet_DR[wp + "-CSVv2"] = VectorUtil::DeltaR(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+        diBJet_DPhi[wp + "-CSVv2"] = VectorUtil::DeltaPhi(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+        diBJet_DEta[wp + "-CSVv2"] = DeltaEta(jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.first], jets.p4[m_diBJet_CSVv2Chosen[wp].idxs.second]);
+      }else{
+        diBJet_DR[wp + "-CSVv2"] = 0;
+        diBJet_DPhi[wp + "-CSVv2"] = 0;
+        diBJet_DEta[wp + "-CSVv2"] = 0;
+      }
+      selectedJets_diBJetExcluded[wp + "-CSVv2"] = std::vector<uint8_t>();
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID_DRcut.size(); ijet++){
+        if(m_diBJet_CSVv2Chosen[wp].idxs.first != ijet && m_diBJet_CSVv2Chosen[wp].idxs.second != ijet)
+          selectedJets_diBJetExcluded[wp + "-CSVv2"].push_back(ijet);
+      }
+
+    }
+
+    ///////////////////////////
+    //    EVENT VARIABLES    //
+    ///////////////////////////
+    
+    // leptons-jets
+   
+    ll_jj_p4 = diLepton_p4 + diJet_p4;
+    ll_jj_DR = VectorUtil::DeltaR(diLepton_p4, diJet_p4);
+    ll_jj_DPhi = VectorUtil::DeltaPhi(diLepton_p4, diJet_p4);
+    ll_jj_DEta = DeltaEta(diLepton_p4, diJet_p4);
+
+    lj_minDR = 10;
+    lj_noDRcut_minDR = 10;
+    if(m_diLepton.idxs.first >= 0){
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID_DRcut.size(); ijet++){
+        float dr = std::min(
+            VectorUtil::DeltaR(m_leptons[m_diLepton.lidxs.first].p4, jets.p4[ijet]),
+            VectorUtil::DeltaR(m_leptons[m_diLepton.lidxs.second].p4, jets.p4[ijet])
+            );
+        if(dr < lj_minDR)
+          lj_minDR = dr;
+      }
+      for(uint8_t ijet = 0; ijet < selectedJets_tightID.size(); ijet++){
+        float dr = std::min(
+            VectorUtil::DeltaR(m_leptons[m_diLepton.lidxs.first].p4, jets.p4[ijet]),
+            VectorUtil::DeltaR(m_leptons[m_diLepton.lidxs.second].p4, jets.p4[ijet])
+            );
+        if(dr < lj_noDRcut_minDR)
+          lj_noDRcut_minDR = dr;
+      }
+    }
+    
+    // leptons-bjets
+
+    for(const std::string &wp: {"LL", "MM", "TT"}){
+      for(const std::string &order: {"-Pt", "-CSVv2"}){
+        ll_bb_p4[wp+order] = diLepton_p4 + diBJet_p4[wp+order];
```

Comment:
(very) small optimization, it may be better to define ``std::string tag = wp + order`` at the beginning to avoid the string concatenation on each line.